libraries/AP_Math/definitions.h

```diff
@@ -117,3 +117,30 @@ static const double WGS84_E = (sqrt(2 * WGS84_F - WGS84_F * WGS84_F));
 
 // Amps microseconds to milliamp hours
 #define AUS_TO_MAH 0.0000002778f
+
+/*
+ * overload fabs so that we get the correct outcome regardless of target type
+ */
+inline float safe_fabs(const float v) {
+    return fabsf(v);
+}
+
+inline double safe_fabs(const double v) {
+#ifdef ALLOW_DOUBLE_MATH_FUNCTIONS
+    return fabs(v);
+#else
+    return fabsf(static_cast<float>(v));
+#endif
+}
+
+inline uint32_t safe_fabs(const uint32_t v) {
+    return v;
+}
+
+inline int32_t safe_fabs(const int32_t v) {
```

Comment:
again, this removes a cast, potentially breaking code

Reply:
Ditto